content/docs/state-and-lifecycle.md

```diff
@@ -191,19 +191,20 @@ ReactDOM.render(
 );
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/KgQpJd?editors=0010)
+[**CodePenda sinab ko'ring**](https://codepen.io/gaearon/pen/KgQpJd?editors=0010)
 
-Next, we'll make the `Clock` set up its own timer and update itself every second.
+Keyin, biz `Clock` komponenti o'zi taymer o'rnatatadigan va har soniyada yangilaydigan qilamiz.
 
 ## Adding Lifecycle Methods to a Class {#adding-lifecycle-methods-to-a-class}
 
-In applications with many components, it's very important to free up resources taken by the components when they are destroyed.
+Ko'p komponentlarga ega ilovalarda, komponentlar yo'q qilinganda ular tomonidan olingabn joy va resurslarni
+bo'shatish juda muhim.
 
-We want to [set up a timer](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval) whenever the `Clock` is rendered to the DOM for the first time. This is called "mounting" in React.
+Biz [taymer o'rnatishni](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval) hohlaymiz qachonki `Clock` komponenti DOMga birinchi marta chizilganda va bu Reactda "mounting" deb ataladi.
 
-We also want to [clear that timer](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/clearInterval) whenever the DOM produced by the `Clock` is removed. This is called "unmounting" in React.
+Shuningdek, biz [taymerni to'xtatishni](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/clearInterval) hohlaymiz qachonki `Clock` komponenti tomonidan yaratilgan DOM olib tashlanganda va bu Reactda "unmounting" deb nomlanadi.
 
-We can declare special methods on the component class to run some code when a component mounts and unmounts:
+Biz komponent o'rnatilganda(mount) va tozalanganda(unmount) biror kod bajarish uchun klass komponentida maxsus metodlarni e'lon qilishimiz mumkin:
```

Comment:
Please use `yechish` for `unmount` [see](https://github.com/reactjs/uz.reactjs.org/blob/main/TRANSLATION.md#:~:text=(un)mount)

Reply:
```suggestion
Biz komponent o'rnatilganda(mount) va yechilganda(unmount) biror kod bajarish uchun klass komponentida maxsus metodlarni e'lon qilishimiz mumkin:
```